product_pricelist_simulation_margin/wizards/wizard_preview_pricelist.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 FactorLibre - Hugo Córdoba
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from odoo import fields, models
+
+from odoo.addons import decimal_precision as dp
+
+
+class WizardPricelistSimulation(models.TransientModel):
+    _inherit = "wizard.preview.pricelist"
+    _description = "wizard - Preview Pricelist"
+
+    def _prepare_simulation_lines_vals(self, variant, pricelist):
+        rslt = super()._prepare_simulation_lines_vals(variant, pricelist)
+        to_currency = pricelist.currency_id
+        res = variant.taxes_id.compute_all(
+            variant.list_price, to_currency, self.product_qty, product=variant
+        )
+        price_vat_excl = res["total_excluded"] / self.product_qty
+        price_vat_incl = res["total_included"] / self.product_qty
+
+        cost = self.env.user.company_id.currency_id._convert(
+            variant.standard_price,
+            to_currency,
+            self.env.user.company_id,
+            self.price_date,
+            round=False,
+        )
+        rslt.update(
+            {
+                "pricelist_id": pricelist.id,
+                "price_vat_excl": price_vat_excl,
+                "price_vat_incl": price_vat_incl,
+                "margin": price_vat_excl - cost,
+                "margin_percent": price_vat_excl
+                and ((price_vat_excl - cost) / price_vat_excl) * 100,
+            }
+        )
+        return rslt
+
+
+class PricelistSimulationLine(models.TransientModel):
+    _inherit = "wizard.preview.pricelist.line"
+    _description = "wizard - Preview Pricelist Line"
+
+    price_vat_excl = fields.Float(string="Unit Sales Price (Excl.)", readonly=True)
+    price_vat_incl = fields.Float(string="Unit Sales Price (Incl.)", readonly=True)
+
+    margin = fields.Float(
+        string="Margin",
+        store=True,
+        digits=dp.get_precision("Product Price"),
+        readonly=True,
+    )
+
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        store=True,
+        digits=dp.get_precision("Product Price"),
+        readonly=True,
+    )
+
+    bg_color = fields.Char(compute="_compute_bg_color")
```

Comment:
I wonder if we should remove bg_color and computation and so one. That was a try in V12, but in fact, the result is not good, and my user asked me to disable the backgrounds.

Not a blocking point. What do you think ?